src/Npgsql/TaskExtensions.cs

```diff
@@ -83,5 +83,27 @@ internal static async Task WithCancellation(this Task task, CancellationToken ca
         internal static Task WithCancellationAndTimeout(this Task task, NpgsqlTimeout timeout, CancellationToken cancellationToken)
             => task.WithCancellation(cancellationToken).WithTimeout(timeout);
 #endif
+
+        internal static TResult ExecuteWithCancellationAndTimeout<TResult>(Func<CancellationToken, TResult> func, NpgsqlTimeout timeout, CancellationToken cancellationToken)
```

Comment:
nit: naming-wise this doesn't actually add any sort of cancellation functionality, only timeout.

Reply:
Fixed.